src/middlewared/middlewared/plugins/system.py

```diff
@@ -152,6 +156,335 @@ def debug(self, job):
+                    f'{schema}.language',
+                    f'Specified "{language}" language not found, kindly correct it'
+                )
+
+        kbdmap_value = data.get('kbdmap')
+        if kbdmap_value:
+            # get kbdmap choices and apply the lower function on them for comparison
+            # this needs work
+            kbdmap_choices = list(map(lambda values: [v.lower() for v in values],
+                                      zip(*await self.get_kbdmap_choices())))
+            '''if not (
+                    kbdmap_value.split('.')[0].lower() in kbdmap_choices[0] or
+                    kbdmap_value.split('.')[0].lower() in kbdmap_choices[1]
+            ):
+                verrors.add(
+                    f'{schema}.kbdmap',
+                    'Please correct kbdmap value'
+                )'''
+
+        timezone = data.get('timezone')
+        if not timezone:
+            verrors.add(
+                f'{schema}.timezone',
+                'This field is required'
+            )
+        else:
+            timezones = await self.get_timezones()
+            if timezone not in timezones:
+                verrors.add(
+                    f'{schema}.timezone',
+                    'Please select a correct timezone'
+                )
+
+        ip4_addresses_list = await self.middleware.call(
+            'interfaces.ipv4_in_use'
+        )
+        ip4_address = data.get('guiaddress')
+        if (
+            ip4_address and
+            ip4_address != '0.0.0.0' and
+            ip4_address not in ip4_addresses_list
+        ):
+            verrors.add(
+                f'{schema}.guiaddress',
+                'Selected ipv4 address is not associated with this machine'
+            )
+
+        ip6_addresses_list = await self.middleware.call(
+            'interfaces.ipv6_in_use'
+        )
+        ip6_address = data.get('guiv6address')
+        if (
+            ip6_address and
+            ip6_address != '::' and
+            ip6_address not in ip6_addresses_list
+        ):
+            verrors.add(
+                f'{schema}.guiv6address',
+                'Selected ipv6 address is not associated with this machine'
+            )
+
+        syslog_level = data.get('sysloglevel')
+        if not syslog_level:
+            verrors.add(
+                f'{schema}.sysloglevel',
+                'This field is required'
+            )
+
+        syslog_server = data.get('syslogserver')
+        if not syslog_server:
+            verrors.add(
+                f'{schema}.syslogserver',
+                'This field is required'
+            )
+        else:
+            match = re.match("^[\w\.\-]+(\:\d+)?$", syslog_server)
+            if not match:
+                verrors.add(
+                    f'{schema}.syslogserver',
+                    'Invalid syslog server format'
+                )
+            elif ':' in syslog_server:
+                port = int(syslog_server.split(':')[-1])
```

Comment:
Probably need to catch a ValueError exception here

Reply:
We wouldn't need that because we already have regex ensuring that all we have is an integer in the if part